moveit_setup_assistant/src/widgets/configuration_files_widget.cpp

```diff
@@ -660,14 +715,17 @@ void ConfigurationFilesWidget::showGenFiles()
     fs::path file_path = config_data_->appendPaths(config_data_->config_pkg_path_, file->rel_path_);
 
     // Checkbox
-    if( file->generate_ )
+    item->setCheckState(file->generate_ ? Qt::Checked : Qt::Unchecked);
+    // externally modified?
+    if (file->modified_)
     {
-      item->setCheckState(Qt::Checked);
-    }
-    else
-    {
-      item->setCheckState(Qt::Unchecked);
-      item->setForeground( QBrush(QColor(255, 135, 0)));
+      if (file->write_on_changes & config_data_->changes)
+      {
+        have_conflicting_changes = true;
+        item->setForeground( QBrush(QColor(255, 0, 0)));
+      }
+      else
+        item->setForeground( QBrush(QColor(255, 135, 0)));
     }
 
```

Comment:
Here and only here, the files entries are colored red or orange. If they are colored red, also the flag `have_conflicting_changes` is set which triggers the extra warning. I cannot see a reason for the behavior observed by @davetcoleman.